test/distributed/fsdp/test_fsdp_mixed_precision.py

```diff
@@ -290,6 +290,33 @@ def _reduce_scatter_base_validate_mp(
 
         return orig_reduce_scatter(*args, **kwargs)
 
+    def _test_grads_reduced_precision(self):
+        class MyModel(nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.lin1 = nn.Linear(10, 10)
+                self.lin2 = nn.Linear(10, 10)
+
+            def forward(self, x):
+                return self.lin2(self.lin1(x))
+
+        m = MyModel().cuda()
+        mp = MixedPrecision(
+            param_dtype=torch.float16,
+            reduce_dtype=torch.float16,
+            buffer_dtype=torch.float16,
+            keep_casted_gradients=True,
+        )
+        m.lin1 = FSDP(m.lin1, mixed_precision=mp)
+        m = FSDP(m, mixed_precision=mp)
+        for _ in range(6):
+            inp = torch.ones(1, 10)
+            m(inp).sum().backward()
+            for param in m.parameters():
+                self.assertEqual(torch.float16, param.grad.dtype)
+
+        dist.barrier()
```

Comment:
Curious: what is the need for the `dist.barrier()` here?

Reply:
I think I was seeing some sort of race where one rank exited early, and the other would NCCL abort in the backward or something like that.